l10n_es_vat_book/models/l10n_es_vat_book.py

```diff
@@ -286,7 +286,7 @@ def _account_move_line_domain(self, taxes, account=None):
         domain = [
             ("date", ">=", self.date_start),
             ("date", "<=", self.date_end),
-            ("parent_state", "=", "posted"),
+            ("move_id.state", "=", "posted"),
```

Comment:
Esto va a redundar en una pérdida de rendimiento. ¿No existe otra opción?

Reply:
El parent_state es de v13, en v12 no existía... habría que ver si hay otra opción, pero por el momento no lo he visto

Reply:
Pero antes de este PR no estaba esa condición, ¿no? Diría que podría considerarse un bug, pero lo solucionaría a otro nivel una vez hecho el search.

Reply:
@pedrobaeza sería al contrario, esa alternativa sería más lenta, el filtrado por la BD siempre va a ser más rápido que hacerlo por el ORM de Odoo.

Reply:
Estoy viendo que el campo `move_id` tiene auto_join=True:

https://github.com/odoo/odoo/blob/720e2daafa9ba686aefc798370b8216e7e8c18e2/addons/account/models/account_move.py#L686

lo que hace que la SQL se genere de un golpe con un LEFT JOIN, lo que hace que no sea excesivamente lento. Si no tuviera eso, se generarían dos consultas SQL, una primera para obtener los account.move con state='posted', y luego la segunda con un `move_id IN ()` y toda la ristra de IDs que cumplen la primera, lo que haría muy lento (aún ejecutándose en SQL).

Lo damos por bueno entonces.